src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -304,7 +305,7 @@ private BoundExpression FixTupleLiteral(ArrayBuilder<DeconstructionVariable> che
                 {
                     var elementSyntax = syntax.Kind() == SyntaxKind.TupleExpression ? ((TupleExpressionSyntax)syntax).Arguments[i] : syntax;
 
-                    hasErrors |= !MakeDeconstructionConversion(tupleOrDeconstructedTypes[i], elementSyntax, rightSyntax, diagnostics,
+                    hasErrors |= !MakeDeconstructionConversion(tupleOrDeconstructedTypes[i], elementSyntax, rightSyntax, LocalScopeDepth, diagnostics,
```

Comment:
It looks like the valEscape we pass here doesn't affect things in practice, but it would be more strictly correct/robust to pass 'rightEscape' here.

Consider a scenario like the following

```cs
    public RS M2(RS a, RS b, RS c, RS d) {
        (a, scoped var bc) = d;
        b = bc;
        return b;
    }
```

The variable we're converting here is essentially 'var bc', and we're declaring it as implicitly 'scoped' by passing 'LocalScopeDepth' as the val escape. But then the way we use it is to immediately assign it out to the variables it's ultimately bound for.